lib/models/service.ts

```diff
@@ -0,0 +1,165 @@
+import * as DocumentDb from "documentdb";
+import * as DocumentDbUtils from "../utils/documentdb";
+import {
+  DocumentDbModelVersioned,
+  IVersionedModel,
+  ModelId
+} from "../utils/documentdb_model_versioned";
+
+import { Option } from "ts-option";
+import { Either } from "../utils/either";
+
+import { FiscalCode } from "../api/definitions/FiscalCode";
+
+import { nonEmptyStringToModelId } from "../utils/conversions";
+import { NonNegativeNumber } from "../utils/numbers";
+import { NonEmptyString } from "../utils/strings";
+
+/**
+ * Base interface for Service objects
+ */
+export interface IService {
+  readonly serviceId: NonEmptyString;
+  // the name of the department within the service
+  readonly departmentName: NonEmptyString;
+  // the name of the service
+  readonly serviceName: NonEmptyString;
+  // the name of the organization
+  readonly organizationName: NonEmptyString;
+  // list of authorized fiscal codes
+  readonly authorizedRecipients?: ReadonlyArray<FiscalCode>;
```

Comment:
this should be a `ReadonlySet` (a recipient is supposed to appear once in the collection)

Reply:
see comment above